File: linkedlist/c/linkedlist.c

#include <stdio.h>
#include <stdlib.h>
struct node {
int data;
struct node* next;
};

typedef struct node node;

node* head =NULL;
node* create_node(int data){
    node* temp = (node *)malloc(sizeof(node));
    temp->data = data;
    temp->next = NULL;

    return temp;
}
void add_node_last(int data){
    node* new_node= create_node(data);
  //  printf("new node is created at heap\n");
    if(head == NULL){
        head = new_node;
        return;
    }
    
    node* temp = head;
    while(temp->next !=NULL){
        temp = temp->next;
    }
    
    temp->next = new_node;
   //  printf("attached to last\n");
}
void traverse_list(){
    node* temp = head;
    while(temp != NULL){
        printf("data %d\n",temp->data);
        temp = temp->next;
    }
}

void insert_at_position(int data, int n){
    node* new_node = create_node(data);

    if(n == 1){
        new_node->next = head;
        head = new_node;
        return;
    }
    node* temp = head;
    
    for (int i = 1; i < n - 1 && temp != NULL; i++) {
        temp = temp->next;
    }

    new_node->next = temp->next;
    temp->next = new_node;
}

void deletion_at_position(int n){
    node* temp = head;
    for(int i=1;i<n-1;i++){
        temp=temp->next;
    }
    node* to_delete = temp->next;
    temp->next = temp->next->next;
    free(to_delete);
}
int main()
{
    add_node_last(2);
    insert_at_position(6,2);
    add_node_last(4);
    deletion_at_position(2);
    traverse_list();

    return 0;
}

File: linkedlist/java/linkedlist.java
import java.io.*;

public class linkedlist{

    Node head;

    static class Node{
        int data;
        Node next;

        Node(int d){
            data = d;
        }
    }

    public static linkedlist insert(linkedlist list,int data){
        Node node = new Node(data);

        if(list.head == null){
            list.head = node;
        }else{
            Node temp = list.head ;

            while(temp.next != null){
                temp = temp.next;
            }
            temp.next = node;
        }
        return list;
    }

    public static void printList(linkedlist list){
        Node temp = list.head;
        while(temp != null){
            System.out.println("data : "+temp.data);
            temp = temp.next;
        }
    }

    public static linkedlist delete(linkedlist list, int key){
        Node temp = list.head;
        Node prev = null;

        if(temp!=null && temp.data == key){
            
            list.head = temp.next;
            return list;  
        }
            while(temp != null && temp.data != key){

                prev = temp;
                temp = temp.next;
            }
            if(temp == null){
               return list;}
            prev.next = temp.next;   
        

        return list;
    }

    public static linkedlist reverse(linkedlist list){
        Node prev, current, nex;
        prev = null;
        current = list.head;

        while(current != null){
            nex = current.next;
            current.next = prev;
            prev = current;
            current = nex;
        }

        list.head = prev;
        return list;
    }

    public static void main(String[] args){

        linkedlist var = new linkedlist();

        var= insert(var,2);
        var= insert(var,4);
        var= insert(var,6);
        var= insert(var,8);
        var = delete(var, 4);
        var= insert(var,2);
        var= insert(var,4);
        var= insert(var,6);
        var= insert(var,8);

        printList(var);
        System.out.println("reverse of the list follows");

        var = reverse(var);

        printList(var);

    }
}